CIP-0048/README.md

```diff
@@ -0,0 +1,249 @@
+- There is no current mechanism to reduce duplicated metadata
+- NFT assets are restricted by there own metadata scope, optional references would prevent this restriction
+- There is currently no way to reference on chain data stored in a different policy
+
+# Specification
+
+### **Inside the 721 metadata JSON**
+
+| reserved | description                                            | scope                     |
+| -------- | ------------------------------------------------------ | ------------------------- |
+| refs     | references tag, defines where to find the onchain data | Inside `<asset_name>` tag |
+| p        | contains raw data (on chain data).                     | Inside `<policy_id>` tag  |
+
+### **The `refs` tag**
+
+Contains the `name`, `mediaType`, `src` tags (and the optional `type` tag)
+
+- `name` (string) is the references name (similar usage to name in the current `files` tag)
+- `mediaType` (string) defines the mime type for the data referenced in `src`
+  - is a string
+- `src` (string array) is an **ordered** array of references.
+  - The order is that of which the data will be parsed. For example `["1","3","2"]` will result in payload `"1"` being the start of the data, followed by payload `"3"` and ending with payload `"2"`.
+
+### **Optional `type` tag for `refs`**
+
+The default CIP-48 behavior is to assume the assets parent `<policy_id>` contains all the payloads.
+
+Optionally we define `type` and two options `policy` and `txhash`. To allow for non-default behavior
+
+- `policy`
+  - Defines an external policy (instead of the default behavior of using the parent policy)
+  - example
+    ```json
+      "refs": [
+        {
+          "name": "...",
+          "mediaType": "...",
+          "type": {
+            "policy": "<some_external_policy>"
+          }
+        }
+      ]
+    ```
+- `txhash`
+  - Defines an array of transaction hashes (instead of the default behavior of using the parent policy)
+  - example
+    ```json
+      "refs": [
+        {
+          "name": "...",
+          "mediaType": "...",
+          "type": {
+            "txhash": ["<mint_txhash>", "<mint_txhash>"]
+          }
+        }
+      ]
+    ```
+  - `txhash` is faster than parsing all transactions in a given policy.
+  - `txhash` can dramatically reduce onchain data usage if duplicated data is referenced here _(however, the bytes required to define the reference would have to be less than the reference data itself)_
+
+### **The payload `p` tag**
+
+- Defined within the `<policy_id>` scope
+- contains payload reference names followed by a string array of payload data
+- example
+  ```json
+    "p":
+      {
+        "1": ["1 is a valid payload"],
+        "payloadA": ["a...64","b...64"], //a string has a max length of 64 characters
+        "payloadB": ["I'm found by referencing payloadB"],
+      }
+  ```
+
+## General structure
+
```

Comment:
It keeps coming up at CIP meetings recently that, since the chain itself stores CBOR (never JSON), that specifications should be written in CDDL.  If this is one of those cases where that's not important then @KtorZ will correct me... and also he has offered to help with translating these specs into CDDL.

We also recommend that the schemas be put in separate files: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0001/README.md#repository-organization